src/Bundle/DependencyInjection/SyliusGridExtension.php

```diff
@@ -77,4 +75,97 @@ public function getConfiguration(array $config, ContainerBuilder $container): Co
 
         return $configuration;
     }
+
+    private function loadPersistence(array $drivers, array $grids, LoaderInterface $loader, ContainerBuilder $container): void
+    {
+        $availableDrivers = $this->getAvailableDrivers($container);
+
+        // Enable all available drivers if there is no configured drivers
+        $drivers = [] !== $drivers ? $drivers : $availableDrivers;
+
+        $gridDrivers = $this->getGridDrivers($grids);
+
+        $this->checkConfiguredDrivers($drivers, $availableDrivers, $gridDrivers);
+
+        foreach ($drivers as $enabledDriver) {
+            if ($enabledDriver === SyliusGridBundle::DRIVER_DOCTRINE_PHPCR_ODM) {
+                @trigger_error(sprintf(
+                    'The "%s" driver is deprecated in Sylius 1.3. Doctrine PHPCR will no longer be supported in Sylius 2.0.',
+                    SyliusGridBundle::DRIVER_DOCTRINE_PHPCR_ODM,
+                ), \E_USER_DEPRECATED);
+            }
+
+            $loader->load(sprintf('services/integrations/%s.xml', $enabledDriver));
+        }
+    }
+
+    /**
+     * @param array<string, array{driver?: array{name: string}}> $drivers
+     *
+     * @return array<string, string>
+     */
+    private function getGridDrivers(array $drivers): array
+    {
+        $gridDrivers = array_map(function (array $grid): string|false {
+            return $grid['driver']['name'] ?? false;
+        }, $drivers);
+
+        // Remove grid with disabled driver
+        return array_filter($gridDrivers, function (string|false $driver): bool {
+            return false !== $driver;
+        });
+    }
+
+    private function getAvailableDrivers(ContainerBuilder $container): array
+    {
+        $availableDrivers = [];
+
+        if ($container::willBeAvailable(SyliusGridBundle::DRIVER_DOCTRINE_ORM, \Doctrine\ORM\EntityManagerInterface::class, ['doctrine/doctrine-bundle'])) {
+            $availableDrivers[] = SyliusGridBundle::DRIVER_DOCTRINE_ORM;
+        }
+
+        if ($container::willBeAvailable(SyliusGridBundle::DRIVER_DOCTRINE_PHPCR_ODM, \Doctrine\ODM\PHPCR\Document\Resource::class, ['doctrine/doctrine-bundle'])) {
+            $availableDrivers[] = SyliusGridBundle::DRIVER_DOCTRINE_PHPCR_ODM;
+        }
+
+        return $availableDrivers;
+    }
+
+    /**
+     * @param string[] $configuredDrivers
+     * @param string[] $availableDrivers
+     * @param array<string, string> $gridDrivers
+     */
+    private function checkConfiguredDrivers(array $configuredDrivers, array $availableDrivers, array $gridDrivers): void
```

Comment:
In validation, `check` is usually reserved for function that returns `:bool` as result, while `assert` is for `:void` with exception.

```suggestion
    private function assertDriversValid(array $configuredDrivers, array $availableDrivers, array $gridDrivers): void
```